src/main/scala/org/broadinstitute/dsde/workbench/sam/model/SamModel.scala

```diff
@@ -84,7 +84,10 @@ object SamResourceActions {
 
 @Lenses case class ResourceTypeName(value: String) extends ValueObject
 
-@Lenses case class Resource(resourceTypeName: ResourceTypeName, resourceId: ResourceId, authDomain: Set[WorkbenchGroupName] = Set.empty)
+@Lenses final case class ResourceIdentity(resourceTypeName: ResourceTypeName, resourceId: ResourceId)
```

Comment:
yea, it is a bit confusing. But I think it's better than `ResourceIdAndTypeName` since that name doesn't indicate well what this field is for. Conceptually, it uniquely identifies a `Resource`, whenever we do equality check or look up for a resource, we need all fields in this case class. https://github.com/broadinstitute/sam/pull/221/files/07b755659f0169b1d1aa8704f0ed08cabbe572a0#diff-a80e785aa2e82ece2775df0a248fde5cR101 is similar idea. I'm not sure if `resourceId` is unique across all resourceTypes, but even if they are, we won't be able to look up a resource just by `resourceId`.

I'm up for other name suggestions tho.